config/state_convert.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+)
+
+// CurrentStateVersion is the current revision for the state file structure
+const CurrentStateVersion = "1.0"
+
+func (state *MixState) parseVersionAndConvert() error {
+	// Reset version for files without versioning
+	state.version = "0.0"
+
+	f, err := os.Open(state.filename)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = f.Close()
+	}()
+
+	// Read state version
+	reader := bufio.NewReader(f)
+	found, err := state.parseVersion(reader)
+	if err != nil {
+		return err
+	}
+
+	// Already on latest version
+	if found && state.version == CurrentStateVersion {
+		return nil
+	}
+
+	fmt.Printf("Converting state to version %s\n", CurrentStateVersion)
+
+	if err := helpers.CopyFile(state.filename+".bkp", state.filename); err != nil {
+		return err
+	}
```

Comment:
This is done for builder.conf because it is a file that holds user config. Since mixer.state should only be modified by mixer commands and is considered and "internal storage", there is no need to do a backup.